app/dashboard/notifications.py

```diff
@@ -277,37 +282,61 @@ def maybe_post_on_craigslist(bounty):
+    for postal_code_input in form.find_all('input', {'id': "postal_code"}):
+        postal_code_input['value'] = '94105'
     form.find('input', {'value': 'pay', 'name': 'remuneration_type'})['checked'] = ''
-    form.find('input', {'id': "remuneration"})['value'] = "{} ETH or {} USD".format(bounty.value_in_eth, bounty.value_in_usdt)
-
+    form.find('input', {'id': "remuneration"})['value'] = "{} ETH".format(bounty.value_in_eth/10**18)
+    try:
+        form.find('input', {'id': "wantamap"})['data-checked'] = ''
+    except:
+        pass
     page = browser.submit(form, form['action'])
 
     # skipping image upload
     form = page.soup.find_all('form')[-1]
     page = browser.submit(form, form['action'])
 
+    for i in range(MAX_URLS):
+        if page.url.endswith('s=preview'):
+            break
+        #Chooses the first default
+        page = browser.submit(form, form['action'])
+        form = page.soup.form
+    else:
+        # for-else magic
+        # if the loop completes normally that means we are still not at the edit page
+        # hence return and don't proceed further
+        return
+
+
     # submitting final form
-    form = page.soup.form
+    form = page.soup.form 
     #getting last email id
-    last_email_id = fetch_last_email_id(settings.IMAP_EMAIl, settings.IMAP_PASSWORD)
+    last_email_id = fetch_last_email_id(settings.IMAP_EMAIL, settings.IMAP_PASSWORD)
     page = browser.submit(form, form['action'])
-    last_email_id_new = fetch_last_email_id(settings.IMAP_EMAIl, settings.IMAP_PASSWORD)
+    time.sleep(10)
+    last_email_id_new = fetch_last_email_id(settings.IMAP_EMAIL, settings.IMAP_PASSWORD)
     #if no email has arrived wait for 5 seconds
     if last_email_id==last_email_id_new:
         # could slow responses if called syncronously in a request
-        import time
         time.sleep(5)
 
-    emails = fetch_mails_since_id( settings.IMAP_EMAIl, settings.IMAP_PASSWORD,last_email_id)
+    emails = fetch_mails_since_id( settings.IMAP_EMAIL, settings.IMAP_PASSWORD,last_email_id)
     for email_id, content in emails.items():
         if 'craigslist' in content['from']:
             for link in re.findall(r"(?:https?:\/\/[a-zA-Z0-9%]+[.]+craigslist+[.]+org/[a-zA-Z0-9\/\-]*)", content.as_string()):
-                #opening all links in the email
-                browser = mechanicalsoup.StatefulBrowser()
-                page = browser.open(link)
-                form = page.soup.form
-                page = browser.submit(form, form['action'])
+                # opening all links in the email
+                try:
+                    browser = mechanicalsoup.StatefulBrowser()
+                    page = browser.open(link)
+                    form = page.soup.form 
+                    page = browser.submit(form, form['action'])
+                    return link
+                except:
+                    # in case of inavalid links
+                    pass
```

Comment:
Absolutely correct. Will fix this

Reply:
Actually, this is fine.
The regex here is a slightly 'loose' regex, so it can match some links which might be invalid. In case of an invalid link we just move ahead and look for further links. We return the link whenever we come across a valid one. At the end, when the loop is completed without us breaking it or returning it again, that means we did not find any valid link. Hence we return false in the 'else; clause